src/parseTools.js

```diff
@@ -488,14 +488,18 @@ function makeReturn64(value) {
   return `(setTempRet0(${pair[1]}), ${pair[0]})`;
 }
 
-function makeThrow(what) {
+function makeThrow(excPtr) {
   if (ASSERTIONS && DISABLE_EXCEPTION_CATCHING) {
-    what += ' + " - Exception catching is disabled, this exception cannot be caught. Compile with -sNO_DISABLE_EXCEPTION_CATCHING or -sEXCEPTION_CATCHING_ALLOWED=[..] to catch."';
+    excPtr += ' + " - Exception catching is disabled, this exception cannot be caught. Compile with -sNO_DISABLE_EXCEPTION_CATCHING or -sEXCEPTION_CATCHING_ALLOWED=[..] to catch."';
     if (MAIN_MODULE) {
-      what += ' + " (note: in dynamic linking, if a side module wants exceptions, the main module must be built with that support)"';
+      excPtr += ' + " (note: in dynamic linking, if a side module wants exceptions, the main module must be built with that support)"';
     }
+    return `throw ${excPtr};`;
+  } else if (EXCEPTION_STACK_TRACES) {
+    return `throw new CppException(${excPtr});`;
+  } else {
```

Comment:
Since each of these blocks returns you don't need any of the `else`s here.  Just two early returns and normal return should work fine,

Reply:
2713a502b959153a56014c2a69c775bcf33cc0bd